diff --git a/build.gradle b/build.gradle
index 1531360..df686d1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -45,7 +45,7 @@ dependencies {
     compile project(':framework')
 
     // ElasticSearch Libraries, including Lucene
-    compile 'org.elasticsearch:elasticsearch:5.6.7' // Apache 2.0
+    compile 'org.elasticsearch:elasticsearch:6.2.4' // Apache 2.0
     // ElasticSearch uses jackson-dataformat-smile, -yaml, -cbor pulling in jackson-core, also used by OrientDB, but gets an older version
     compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-smile:2.9.5'
     compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.9.5'
@@ -53,10 +53,13 @@ dependencies {
     // JNA, needed by ElasticSearch when running embedded, now included as a dependency: compile 'net.java.dev.jna:jna:4.5.0' // Apache 2.0
     // not currently needed/used: compile 'net.java.dev.jna:jna-platform:4.5.0' // Apache 2.0
 
+    // plugin-classloader now required as of ES 6.2
+    compile 'org.codelibs.elasticsearch.lib:plugin-classloader:6.2.4'
+
     // modules: when updating these update the versions in the plugin-descriptor.properties files in home/modules/*
-    compile module('org.codelibs.elasticsearch.module:lang-groovy:5.6.7')
-    compile 'org.codelibs.elasticsearch.module:transport-netty4:5.6.7'
-    compile 'org.codelibs.elasticsearch.module:reindex:5.6.7'
+    // lang-groovy no longer supported as of ES 6.0.0: compile module('org.codelibs.elasticsearch.module:lang-groovy:5.6.7')
+    compile 'org.codelibs.elasticsearch.module:transport-netty4:6.2.4'
+    compile 'org.codelibs.elasticsearch.module:reindex:6.2.4'
 
     // painless script needs antlr4, asm-all (if added put in framework and do modules for pegdown/parboiled so it it shared)
     // problem with lang-painless: 450KB jar file, can't come from jcenter so far, so using groovy in spite of warnings
diff --git a/home/config/elasticsearch.yml b/home/config/elasticsearch.yml
index 66a1303..8cdf949 100644
--- a/home/config/elasticsearch.yml
+++ b/home/config/elasticsearch.yml
@@ -60,7 +60,7 @@ node.ingest: true
 network.host:
   - _site_
   - _local_
-transport.type: local
+# removed as of ES 6.0.0: transport.type: local
 discovery.type: single-node
 
 # CORS settings for local testing only
@@ -114,8 +114,9 @@ http.port: 9200
 script.allowed_types: inline
 script.allowed_contexts: search, aggs
 
-script.engine.groovy.inline.search: true
-script.engine.groovy.inline.aggs: true
+# no more groovy script as of 6.0.0:
+# script.engine.groovy.inline.search: true
+# script.engine.groovy.inline.aggs: true
 
 # moqui-elasticsearch now uses bulk indexing for everything, so use a bigger queue
 # default is 50 for bulk queue, default is 200 for index queue
diff --git a/home/modules/reindex/plugin-descriptor.properties b/home/modules/reindex/plugin-descriptor.properties
index 226f9df..6ea0730 100644
--- a/home/modules/reindex/plugin-descriptor.properties
+++ b/home/modules/reindex/plugin-descriptor.properties
@@ -22,7 +22,7 @@
 description=The Reindex module adds APIs to reindex from one index to another or update documents in place.
 #
 # 'version': plugin's version
-version=5.6.7
+version=6.2.4
 #
 # 'name': the plugin name
 name=reindex
@@ -37,7 +37,7 @@ classname=org.elasticsearch.index.reindex.ReindexPlugin
 java.version=1.8
 #
 # 'elasticsearch.version': version of elasticsearch compiled against
-elasticsearch.version=5.6.7
+elasticsearch.version=6.2.4
 ### optional elements for plugins:
 #
 # 'has.native.controller': whether or not the plugin has a native controller
diff --git a/home/modules/transport-netty4/plugin-descriptor.properties b/home/modules/transport-netty4/plugin-descriptor.properties
index 65e5ada..1a257c4 100644
--- a/home/modules/transport-netty4/plugin-descriptor.properties
+++ b/home/modules/transport-netty4/plugin-descriptor.properties
@@ -22,7 +22,7 @@
 description=Netty 4 based transport implementation
 #
 # 'version': plugin's version
-version=5.6.7
+version=6.2.4
 #
 # 'name': the plugin name
 name=transport-netty4
@@ -37,7 +37,7 @@ classname=org.elasticsearch.transport.Netty4Plugin
 java.version=1.8
 #
 # 'elasticsearch.version': version of elasticsearch compiled against
-elasticsearch.version=5.6.7
+elasticsearch.version=6.2.4
 ### optional elements for plugins:
 #
 # 'has.native.controller': whether or not the plugin has a native controller
diff --git a/service/org/moqui/search/SearchServices.xml b/service/org/moqui/search/SearchServices.xml
index 010fa6f..ab3c707 100644
--- a/service/org/moqui/search/SearchServices.xml
+++ b/service/org/moqui/search/SearchServices.xml
@@ -321,7 +321,7 @@ along with this software (see the LICENSE.md file). If not, see
 
             SearchHits hits = srb.execute().actionGet().getHits()
             for (SearchHit hit in hits) {
-                Map document = hit.getSource()
+                Map document = hit.getSourceAsMap()
                 if (flattenDocument) document = flattenNestedMap(document)
 
                 // As of ES 2.0 _index, _type, _id aren't included in the document
